crates/net/network/src/transactions/mod.rs

```diff
@@ -1260,7 +1240,20 @@ where
             let acc = &mut poll_durations.acc_tx_events;
             duration_metered_exec!(
                 {
-                    // advance incoming transaction events
+                    // Advance incoming transaction events (stream new txns/announcements from
+                    // network manager and queue for import to pool/fetch txns).
+                    //
+                    // This will potentially remove hashes from hashes pending fetch, it the event
+                    // is an announcement (if same hashes are announced that didn't fit into a
+                    // previous request).
+                    //
+                    // If this is an event with `Transactions` message, since transactions aren't
+                    // validated until they are inserted into the pool, this can potentially queue
+                    // >13k transactions for insertion to pool. More if the message size is bigger
+                    // than the soft limit on a `Transactions` broadcast message, which is 128 KiB.
+                    //
+                    // The smallest decodable transaction is an empty legacy transaction, 10 bytes.
+                    // 128 KiB / 10 bytes > 13k transactions
```

Comment:
I still think:

> This section should be structured differently, starting from the soft limit so that it's immediately clear that the limiting factor is the size of the message.

Reply:
I can do that. already capitalised first letter of each sentence and moved the "since" part.